ENIGMAsystem/SHELL/Platforms/SDL/Window.cpp

```diff
@@ -0,0 +1,254 @@
+
+void destroyWindow() {
+  SDL_DestroyWindow(windowHandle);
+  SDL_Quit();
+}
+
+int handleEvents() { return eventHandler.processEvents(); }
+
+}  // namespace enigma
+
+namespace enigma_user {
+
+void io_handle() {}
+
+int window_get_visible() {
+  Uint32 flags = SDL_GetWindowFlags(windowHandle);
+  return (flags & SDL_WINDOW_SHOWN);
+}
+
+void window_set_visible(bool visible) { (visible) ? SDL_ShowWindow(windowHandle) : SDL_HideWindow(windowHandle); }
+
+std::string window_get_caption() { return SDL_GetWindowTitle(windowHandle); }
+
+void window_set_caption(std::string caption) { SDL_SetWindowTitle(windowHandle, caption.c_str()); }
+
+int window_mouse_get_x() {
+  int x;
+  SDL_GetMouseState(&x, nullptr);
+  return x;
+}
+
+int window_mouse_get_y() {
+  int y;
+  SDL_GetMouseState(nullptr, &y);
+  return y;
+}
+
+void window_mouse_set(int x, int y) { SDL_WarpMouseInWindow(windowHandle, x, y); }
+
+int display_mouse_get_x() {
+  int x;
+  SDL_GetGlobalMouseState(&x, nullptr);
+  return x;
+}
+
+int display_mouse_get_y() {
+  int y;
+  SDL_GetGlobalMouseState(nullptr, &y);
+  return y;
+}
+
+void display_mouse_set(int x, int y) { SDL_WarpMouseGlobal(x, y); }
+
+bool window_get_stayontop() {
+  Uint32 flags = SDL_GetWindowFlags(windowHandle);
+  return (flags & SDL_WINDOW_ALWAYS_ON_TOP);
+}
+
+void window_set_stayontop(bool stay) {
+  /* SDL (as of 2.0.5) can set SDL_WINDOW_ALWAYS_ON_TOP 
+  *  upon creation but not after. Also, this only works 
+  * in X11
+  */
+}
+
+bool window_get_sizeable() {
+  Uint32 flags = SDL_GetWindowFlags(windowHandle);
+  return (flags & SDL_WINDOW_RESIZABLE);
+}
+
+void window_set_sizeable(bool resize) { SDL_SetWindowResizable(windowHandle, (resize) ? SDL_TRUE : SDL_FALSE); }
+
+bool window_get_showborder() {
+  Uint32 flags = SDL_GetWindowFlags(windowHandle);
+  return !(flags & SDL_WINDOW_BORDERLESS);
+}
+
+void window_set_showborder(bool show) { SDL_SetWindowBordered(windowHandle, (show) ? SDL_TRUE : SDL_FALSE); }
+
+bool window_get_minimized() {
+  Uint32 flags = SDL_GetWindowFlags(windowHandle);
+  return (flags & SDL_WINDOW_MINIMIZED);
+}
+
+void window_set_minimized(bool min) {
+  if (min != window_get_minimized()) (min) ? SDL_MinimizeWindow(windowHandle) : SDL_RestoreWindow(windowHandle);
+}
+
+bool window_get_maximized() {
+  Uint32 flags = SDL_GetWindowFlags(windowHandle);
+  return (flags & SDL_WINDOW_MAXIMIZED);
+}
+
+void window_set_maximized(bool max) {
+  if (max != window_get_maximized()) (max) ? SDL_MaximizeWindow(windowHandle) : SDL_RestoreWindow(windowHandle);
+}
+
+int window_get_x() {
+  int x;
+  SDL_GetWindowPosition(windowHandle, &x, nullptr);
+  return x;
+}
+
+int window_get_y() {
+  int y;
+  SDL_GetWindowPosition(windowHandle, nullptr, &y);
+  return y;
+}
+
+void window_set_position(int x, int y) { SDL_SetWindowPosition(windowHandle, x, y); }
+
+void window_set_rectangle(int x, int y, int w, int h) {
+  window_set_size(w, h);
+  window_set_position(x, y);
```

Comment:
Ugh, SDL, why? This is going to cause agitating flickering on some systems as the window moves around. These calls are probably arbitrarily synchronous on various platforms.

Reply:
Don't know. I didn't see a all in one function in their API if it's real issue we can just submit a patch to them